packages/x-data-grid/src/hooks/core/pipeProcessing/useGridPipeProcessing.ts

```diff
@@ -119,18 +126,16 @@ export const useGridPipeProcessing = (apiRef: React.MutableRefObject<GridPrivate
     GridPipeProcessingApi['unstable_applyPipeProcessors']
   >((...args) => {
     const [group, value, context] = args as [GridPipeProcessorGroup, any, any];
-    if (!processorsCache.current[group]) {
+    if (!cache.current[group]) {
       return value;
     }
 
-    const preProcessors = Array.from(processorsCache.current[group]!.processors.values());
-    return preProcessors.reduce((acc, preProcessor) => {
-      if (!preProcessor) {
-        return acc;
-      }
-
-      return preProcessor(acc, context);
-    }, value);
+    const processors = cache.current[group]!.processorsAsArray;
+    let result = value;
+    for (let i = 0; i < processors.length; i += 1) {
+      result = processors[i](result, context);
+    }
+    return result;
```

Comment:
This block is expensive. `Map -> Iterator object -> Array.from() -> array.reduce` is costly when this is run 2x for each cell.